app/src/main/java/com/github/se/cyrcle/model/review/ReviewViewModel.kt

```diff
@@ -83,41 +100,81 @@ class ReviewViewModel(
       Log.e("ReviewViewModel", "No review selected")
       return
     }
+
     reviewRepository.addReport(
         report,
+        // If the report was successfully added, check if the associated object is already in
+        // reported objects
         onSuccess = {
-          // Check thresholds for reporting to `ReportedObjectRepository`
-          // If it's a max severity report, add it to the maximum severity count and check if it
-          // exceeds the maximum severity threshold. Otherwise, check if it exceeds the standard
-          // threshold.
-          if ((report.reason.severity == MAX_SEVERITY &&
-              selectedReview.nbMaxSeverityReports >= NB_REPORTS_MAXSEVERITY_THRESH) ||
-              (selectedReview.nbReports >= NB_REPORTS_THRESH)) {
-            reportedObjectRepository.addReportedObject(
-                ReportedObject(
-                    objectUID = selectedReview.uid,
-                    reportUID = report.uid,
-                    reason = ReportReason.Review(report.reason),
-                    userUID = user.public.userId,
-                    objectType = ReportedObjectType.REVIEW),
-                onSuccess = {},
-                onFailure = {})
-          }
-
-          // Update the local reports and review metrics
-          if (report.reason.severity == MAX_SEVERITY) {
-            selectedReview.nbMaxSeverityReports += 1
-          }
-          selectedReview.nbReports += 1
-          reviewRepository.updateReview(selectedReview, {}, {})
-          _selectedReviewReports.update { currentReports -> currentReports?.plus(it) ?: listOf(it) }
-          Log.d("ReviewViewModel", "Report added successfully")
+          val newReportedObject =
+              ReportedObject(
+                  objectUID = selectedReview.uid,
+                  reportUID = report.uid,
+                  nbOfTimesReported = selectedReview.nbReports + 1,
+                  nbOfTimesMaxSeverityReported =
+                      if (report.reason.severity == 3) selectedReview.nbMaxSeverityReports + 1
+                      else selectedReview.nbMaxSeverityReports,
+                  userUID = user.public.userId,
+                  objectType = ReportedObjectType.REVIEW)
+          Log.d("AAAAAAAAAAAAAAA", selectedReview.nbReports.toString())
```

Comment:
Nice `Log` tag :) But maybe a more relevant one would be nice

Reply:
Oops, since I left it as draft I forgot to remove then!